### PR TITLE
📖update architecture doc per current output

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

some of the existing doc output is incorrect per test result

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
Thanks for the PR @jichenjc! Please fix the markdown issue, and we can trigger mandatory integration test. Thank you.

**Comment by jichenjc** [comment]
@furkatgofurov7 thanks~, updated

**Comment by furkatgofurov7** [comment]
@jichenjc I just noticed there are few more places in the same documentation where it needs an update in terms of API versions(all CAPI objects should have `v1alpha4` and CAPM3 objects `v1alpha5` API versions). It would be great, if you could take care of updating those in the same PR.

**Comment by jichenjc** [comment]
@furkatgofurov7 updated again :)

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]
Thanks, there is only one mention of API for Metal3Cluster that is left still in line 131 that needs an update.

**Review by fmuyassarov — APPROVED** [review]

File: docs/architecture.md [hunk 1]
```diff
@@ -313,18 +315,18 @@ spec:
...
   |  name: test1-controlplane-0                                                |
   |  namespace: metal3                                                         |
-  |# Image comes from 'Metal3Machine' and is added by 'CAPM3'               |
+  |# Image comes from 'Metal3Machine' and is added by 'CAPM3'                  |
   |image:                                                                      |
   |  checksum: http://172.22.0.1/images/bionic-server-cloudimg-amd64.img.md5sum|
   |  url: http://172.22.0.1/images/bionic-server-cloudimg-amd64.img            |
   |online: true                                                                |
-  |# UserData comes from 'Metal3Machine' and is added by 'CAPBK'            |
+  |# UserData comes from 'Metal3Machine' and is added by 'CAPBK'               |
```
Comment (kashifest) [inline docs/architecture.md:329]: Is it still called CAPBK?

**Review by kashifest — APPROVED** [review]

**Pull request merged**